internal/executor/executor_test.go

```diff
@@ -47,6 +49,39 @@ func executorTypes() map[string]executorConstructor {
 	}
 }
 
+// pipeReader avoids data races when reading from the docker executor stdout.
+// These races happen if you just use a byte buffer for stdout since there isn't
+// synchronization writing to the byte buffer in the docker executor
+type pipeReader struct {
+	buffer bytes.Buffer
+	reader io.Reader
+	writer io.WriteCloser
+	wg     sync.WaitGroup
+}
+
+func (p *pipeReader) start(t *testing.T) {
+	p.wg.Add(1)
+	go func() {
+		result, err := io.ReadAll(p.reader)
+		p.buffer.Write(result)
```

Comment:
It seems like this should log the error (if any) so that it's trackable during test runtime.

Reply:
I updated all of these to assert no error. Does that seem reasonable?